### PR TITLE
Add LIP "Define state and state transitions of Token module"

### PR DESCRIPTION
**Pull request by MaximeGagnebin**

### CODE REVIEW

File: proposals/lip-define_state_and_state_transition_of_token_module.md [hunk 1]
```diff
@@ -0,0 +1,1287 @@
...
+
+#### Logic from Other Modules
+
+Calling a function `fct` implemented in the [Interoperability module][base-interoperability-LIP] is represented by `interoperability.fct(required inputs)`.
+
+
+### Token Identification
+
+All tokens in the ecosystem are identified by a pair of non-negative integers `(chainID, localID)`, both strictly less than 2^32 . 
+The first element of the pair, `chainID`, is the chain ID of the chain that minted the token (an integer, as specified in the ["Chain Registration" LIP][registration-lip]) and
```
Comment (IkerAlus) [inline proposals/lip-define_state_and_state_transition_of_token_module.md:255]: update the LIP name to "Introduce chain registration mechanism"

**Review by IkerAlus — CHANGES_REQUESTED** [review]
One little fix regarding A LIP title, the rest LGTM.

**Review by IkerAlus — APPROVED** [review]

File: proposals/lip-define_state_and_state_transition_of_token_module.md [hunk 2]
```diff
@@ -0,0 +1,1287 @@
...
+
+```java
+supplyStoreSchema = {
+    "type": "object",
+    "required": ["totalSupply"],
+    "properties": {
+        "totalSupply": { 
+            "dataType": "uint64",
+            "fieldNumber": 1
+        },
```
Comment (janhack) [inline proposals/lip-define_state_and_state_transition_of_token_module.md:308]:
```suggestion
        }
```

File: proposals/lip-define_state_and_state_transition_of_token_module.md [hunk 3]
```diff
@@ -0,0 +1,1287 @@
...
+```java
+availableLocalIDStoreSchema = {
+    "type": "object",
+    "properties": {
+        "nextAvailableLocalID": { 
+            "dataType": "uint32",
+            "fieldNumber": 1
+        },
+    },
+    "required": ["nextAvailableLocalID"]
```
Comment (janhack) [inline proposals/lip-define_state_and_state_transition_of_token_module.md:333]: This schema and the following two were not changed. Note also that some closing commas need to be removed here and for the following two.

**Review by janhack — APPROVED** [review]
I left some very minor comments regarding some schemata.

Nice work Maxime!

File: proposals/lip-define_state_and_state_transition_of_token_module.md (hunk 2 above)
Comment (MaximeGagnebin) [reply proposals/lip-define_state_and_state_transition_of_token_module.md]: thanks done

File: proposals/lip-define_state_and_state_transition_of_token_module.md (hunk 3 above)
Comment (MaximeGagnebin) [reply proposals/lip-define_state_and_state_transition_of_token_module.md]: thanks done

**Review by ManuGowda — APPROVED** [review]

**Review by mitsuaki-u — APPROVED** [review]

**Review by karmacoma — APPROVED** [review]

**Pull request merged**